unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/util/StringJoinerTest.scala

```diff
@@ -0,0 +1,193 @@
+      new StringJoiner(", ", "[", "]")
+        .add("one")
+        .add("two")
+        .add("three")
+        .setEmptyValue("--")
+    }
+    assertJoinerResult("") {
+      new StringJoiner(",").setEmptyValue("--").add("")
+    }
+    assertJoinerResult("one,") {
+      new StringJoiner(",").setEmptyValue("--").add("one").add("")
+    }
+  }
+
+  @Test def testMerge(): Unit = {
+    val empty = new StringJoiner(";", "[", "]").setEmptyValue("--")
+    val single =
+      new StringJoiner(";", "[", "]").setEmptyValue("--").add("single")
+    val multiple = new StringJoiner(";", "[", "]")
+      .setEmptyValue("--")
+      .add("a")
+      .add("b")
+      .add("c")
+    val singleBlank =
+      new StringJoiner(";", "[", "]").setEmptyValue("--").add("")
+
+    assertJoinerResult("+++") {
+      new StringJoiner(", ", "{", "}").merge(empty).setEmptyValue("+++")
+    }
+    assertJoinerResult("+++") {
+      new StringJoiner(", ", "{", "}").setEmptyValue("+++").merge(empty)
+    }
+    assertJoinerResult("{}") {
+      new StringJoiner(", ", "{", "}").merge(singleBlank).setEmptyValue("+++")
+    }
+    assertJoinerResult("{}") {
+      new StringJoiner(", ", "{", "}").setEmptyValue("+++").merge(singleBlank)
+    }
+    assertJoinerResult("{one, two}") {
+      new StringJoiner(", ", "{", "}").add("one").merge(empty).add("two")
+    }
+    assertJoinerResult("{one, single, two}") {
+      new StringJoiner(", ", "{", "}").add("one").merge(single).add("two")
+    }
+    assertJoinerResult("{one, a;b;c, two}") {
+      new StringJoiner(", ", "{", "}").add("one").merge(multiple).add("two")
+    }
+    assertJoinerResult("{one, , two}") {
+      new StringJoiner(", ", "{", "}").add("one").merge(singleBlank).add("two")
+    }
+    assertJoinerResult("{single}") {
+      new StringJoiner(", ", "{", "}").merge(single)
+    }
+    assertJoinerResult("{a;b;c}") {
+      new StringJoiner(", ", "{", "}").merge(multiple)
+    }
+    assertJoinerResult("{}") {
+      new StringJoiner(", ", "{", "}").merge(singleBlank)
+    }
+  }
+
+  @Test def testState(): Unit = {
+    val mutableCharSeq = CharBuffer.allocate(2).put(0, '?').put(1, '!')
+
+    val joiner =
+      new StringJoiner(mutableCharSeq, mutableCharSeq, mutableCharSeq)
+    assertJoinerResult("?!?!")(joiner)
+    joiner.setEmptyValue(mutableCharSeq)
+    assertJoinerResult("?!")(joiner)
+
+    mutableCharSeq.put(0, '-')
+    assertJoinerResult("?!")(
+      joiner
+    ) // the previously set emptyValue is not affected
+    joiner.setEmptyValue(mutableCharSeq)
+    assertJoinerResult("-!")(joiner)
+
+    joiner.add("one")
+    assertJoinerResult("?!one?!")(
+      joiner
+    ) // the previously set prefix and suffix are not affected
+
+    joiner.add("two")
+    assertJoinerResult("?!one?!two?!")(
+      joiner
+    ) // the previously set delimiter is not affected
+  }
+
+  @Test def testNPE(): Unit = {
+
+    /** Both Scala Native and Scala JVM has compliant null pointers. The
```

Comment:
Nice handling of complexity.

A native speaker of English would probably use "have" instead of "has".  Not worth a CI run, leave some room for improvement for the next generation.

The content is what is important, IMO.